src/web_app/js/peerconnectionclient.js

```diff
@@ -358,7 +361,100 @@ PeerConnectionClient.prototype.onRemoteStreamAdded_ = function(event) {
 };
 
 PeerConnectionClient.prototype.onError_ = function(tag, error) {
+  var localSdp = this.localSdp || null;
+  var remoteSdp = this.remoteSdp || null;
+  // Enumerate supported callstats error types.
+  // http://www.callstats.io/api/#enumeration-of-wrtcfuncnames
+  var webrtcFuncNames = {
+    getUserMedia: 'getUserMedia',
+    createOffer: 'createOffer',
+    createAnswer: 'createAnswer',
+    setLocalDescription: 'setLocalDescription',
+    setRemoteDescription: 'setRemoteDescription',
+    addIceCandidate: 'addIceCandidate'
+  };
+
   if (this.onerror) {
     this.onerror(tag + ': ' + error.toString());
+
+    // Only report supported error types to the callstats backend.
+    if (webrtcFuncNames[tag]) {
+      this.callStats_.reportError(this.pc_, this.conferenceId,
+        webrtcFuncNames[tag], error, localSdp, remoteSdp);
+    }
+  }
+};
+
+// Setup the callstats api and attach it to the peerconnection.
+PeerConnectionClient.prototype.setupCallStats_ = function() {
+  // Check dependencies.
+  if (typeof $ !== 'function' && typeof io !== 'function' &&
+      typeof jsSHA !== 'function')  {
+    trace('Callstats dependencies missing, stats will not be setup.');
+    return;
+  }
+
+  // Need to catch the error otherwise the peerConnection creation
+  // will fail.
+  try {
+    trace('Setting up callstats.');
+    // jscs:disable requireCapitalizedConstructors
+    /* jshint newcap: false */
+    var callStats = new callstats(null, io, jsSHA);
+    // jscs:enable requireCapitalizedConstructors
+    /* jshint newcap: true */
+
+    var appId = this.params_.callStatsParams.appId;
+    var appSecret = this.params_.callStatsParams.appSecret;
+    var userId = this.params_.clientId;
+    var conferenceId = this.params_.roomId;
+    // We do not know the remote client id. Setting NA for now.
+    var remoteUserId = 'NA';
+    // Multiplex should be used when sending audio and video on a peerConnection.
+    // http://www.callstats.io/api/#enumeration-of-fabricusage
+    // TODO: Might need to change this dynamically if an audio/video only call.
+    var usage = callStats.fabricUsage.multiplex;
+    var statsCallback = null;
+    var configParams = null;
+    var callback = function(status, msg) {
+      trace('Callstats status: ' + status + ' msg: ' + msg);
+    };
+
+    // Init the callstats api.
+    callStats.initialize(appId, appSecret, userId, callback, statsCallback,
+      configParams);
+
+    // Hookup the callstats api to the peerconnection object.
+    callStats.addNewFabric(this.pc_, remoteUserId, usage,
+      conferenceId, callback);
```

Comment:
Why is the remoteuserId passed as NA? It should pass the actual remoteUserId because it has dependency in dashboard to view the connection between each participant pairs. If its unavailable at this point, addNewFabric can be called before createOffer/createAnswer